src/redis_cmd.cc

```diff
@@ -4058,22 +4058,42 @@ class CommandFetchMeta : public Commander {
   }
 
   Status Execute(Server *svr, Connection *conn, std::string *output) override {
-    uint64_t file_size;
-    rocksdb::BackupID meta_id;
-    int fd;
-    auto s = Engine::Storage::BackupManager::OpenLatestMeta(
-        svr->storage_, &fd, &meta_id, &file_size);
-    if (!s.IsOK()) {
-      LOG(ERROR) << "Failed to open latest meta, err: " << s.Msg();
-      return Status(Status::DBBackupFileErr, "can't create db backup");
-    }
-    // Send the meta ID
-    conn->Reply(std::to_string(meta_id) + CRLF);
-    // Send meta file size
-    conn->Reply(std::to_string(file_size) + CRLF);
-    // Send meta content
-    conn->SendFile(fd);
+    int repl_fd = conn->GetFD();
+    std::string ip = conn->GetIP();
+
+    Util::SockSetBlocking(repl_fd, 1);
+    conn->NeedNotClose();
+    conn->EnableFlag(Redis::Connection::kCloseAsync);
     svr->stats_.IncrFullSyncCounter();
+
+    // Feed-replica-meta thread
+    std::thread t = std::thread([svr, repl_fd, ip]() {
+      Util::ThreadSetName("feed-replica-meta");
+      int fd;
+      uint64_t file_size;
+      rocksdb::BackupID meta_id;
+      auto s = Engine::Storage::BackupManager::OpenLatestMeta(
+          svr->storage_, &fd, &meta_id, &file_size);
+      if (!s.IsOK()) {
+        close(repl_fd);
```

Comment:
if we close without sending error msg to replica, the replica can't log the reason why master close the connection, seems like with a error msg will be better

Reply:
Yes, thanks